flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/FactoryUtil.java

```diff
@@ -148,11 +150,27 @@ public static DynamicTableSink createTableSink(
 			CatalogTable catalogTable,
 			ReadableConfig configuration,
 			ClassLoader classLoader) {
+		return createTableSink(catalog, objectIdentifier, catalogTable, configuration, classLoader, false);
+	}
+
+	/**
+	 * Creates a {@link DynamicTableSink} from a {@link CatalogTable}.
+	 *
+	 * <p>It considers {@link Catalog#getFactory()} if provided.
+	 */
+	public static DynamicTableSink createTableSink(
```

Comment:
Not only source, sink too.
I can see the bug in `HiveDynamicTableFactory` because not remove old method.